service/protocol.hpp

```diff
@@ -19,7 +19,8 @@ namespace raven
     unknown_id,
     unknown_key,
     unknown_setting,
-    unknown_alias
+    unknown_alias,
+    db_error = -1
```

Comment:
Any reason for the negative id?

Reply:
I think you're doing this in order to be able to return it as a config id. I'd question the fact that it is a request state in that case.

Reply:
Perfectly right, negative value is only for this case

Reply:
In case of error we get an request_state set to error_db otherwise the column_id used as a CONFIG_KEY

Reply:
I think it's minor anyway, merging as is.